### PR TITLE
Special case the global OpenShift image pull secret

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

This PR is an enhancement of #2572 and takes into account existing global pull secrets.

[Openshift offers a global pull secret](https://docs.openshift.com/container-platform/4.11/openshift_images/managing_images/using-image-pull-secrets.html#images-update-global-pull-secret_using-image-pull-secrets) named `pull-secret` in the `openshift-config` namespace.

This pull secret can be used to give access to registries on a global scale, with the pull secret being rolled out to each nodes respective container runtime configuration.

The PR will make sure that the global pull secret will be added to the pull secrets for a specific image, irrespective of whether it's not matching in e.g. namespace.

## Checklist
- [x] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

## Testing Performed

- see CI.

### CODE REVIEW

**Comment by dhaus67** [comment]
@misberner: Most of the changes are from #2572 , I already reviewed that one prior.
If you have the time, it might be worth for you to have a look as well.

I'll refactor this PR to _not_ include the changes from #2572 so its better to review, would appreciate a second look from you as well.

**Review by dhaus67 — APPROVED** [review]

File: pkg/images/enricher/enricher_impl.go [hunk 1]
```diff
@@ -655,6 +658,10 @@ func (e *enricherImpl) checkRegistryForImage(image *storage.Image) error {
 	return nil
 }
 
+func isOpenshiftGlobalRegistry(source *storage.ImageIntegration_Source) bool {
```
Comment (misberner) [inline pkg/images/enricher/enricher_impl.go:661]:
```suggestion
func isOpenshiftGlobalPullSecret(source *storage.ImageIntegration_Source) bool {
```
the pull secret has nothing to do with an OpenShift registry, at least not necessarily so

**Review by misberner — APPROVED** [review]

**Comment by dhaus67** [comment]
CI flakes are addressed with #3471 , so merging this.

**Pull request merged**